src/main/java/internet/shop/dao/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package internet.shop.dao.impl;
+
+import internet.shop.dao.ProductDao;
+import internet.shop.db.Storage;
+import internet.shop.lib.Dao;
+import internet.shop.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+    @Override
+    public Product createProduct(String name, Double price) {
+        return new Product(name, price);
+    }
+
+    @Override
+    public void addProduct(Product product) {
+        Storage.addProduct(product);
+    }
+
+    @Override
+    public Optional<Product> getProduct(Long id) {
+        return Storage.products.stream()
+                .filter(x -> x.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public boolean removeProduct(Long id) {
+        return Storage.products.remove(getProduct(id).orElseThrow());
+    }
+
+    @Override
+    public List<Product> getAllProducts() {
+        return Storage.products;
+    }
+
+    @Override
+    public void updateProduct(Product product) {
+        Storage.products.stream()
+                .filter(x -> x.getName().equals(product.getName()))
+                .forEach(x -> x.setPrice(product.getPrice()));
+    }
```

Comment:
Maybe better to compare the product`s bu id. And try to rewrite the whole object because if we'll add some new fields to Product you'll need to change this part of the code too.

Reply:
We can implement it however we like :)

Reply:
product name can be the same in different products and also you may have changed the  name of product and want to update it in Storage.
Let's use Id to find some item and then replace it in Storage with a new one